Folder/asset_data.py

```diff
@@ -0,0 +1,39 @@
+import os
+import requests
+from dotenv import load_dotenv
+
+
+class DataManager:
+    def __init__(self):
+
+        self.cg_endpoint = "https://api.coingecko.com/api/v3"
+        self.crypto_data = {}
+        self.global_data = {}
+        self.cg_header = {
+            "x_cg_demo_api_key": os.environ["CG_API_KEY"],
+        }
+
+    def update_price_data(self):
+
+        cg_params = {
+            "ids": "bitcoin,ethereum,solana,dogecoin",
+            "vs_currencies": "usd",
+            "include_market_cap": "true",
+            "include_24hr_change": "true",
+            "include_24h_vol": "true",
+            "include_last_updated_at": "true",
+        }
+
+        crypto_response = requests.get(url=f"{self.cg_endpoint}/simple/price", params=cg_params, headers=self.cg_header)
+        crypto_response.raise_for_status()
+        self.crypto_data = crypto_response.json()
```

Comment:
you can have this block as its own function, that hits the `simple/price` endpoint. A function having a singular job makes things easier to work with

Reply:
you have the same code being used in another file, I might wrap it like 
```
class Requesteroni:
     def __init__(self, headers: Optional[dict] = None):
           self.headers = headers
     def get(self, url, params = None):
          response = requests.get(url, params=params, headers=self.headers)
          response.raise_for_status()
          return response.json()
```

and then use like 
```
    cheeseeeee = Requseteroni(headers = self.cg_headers)
    self.crypto_data = cheeese.get(f"{self.cg_endpoint"}/simple/price, params)
    self.global_data = cheeeseeeee.get(f"{self.cg_endpoint"}/global, params)

Reply:
This class would just go in asset_data.py, yeah? The convention i've been following has just been like one class each .py file cause i'm a noob n stuff (and that's the format the lessons follow), but these two classes seem like they would work in tangent.

I was thinking earlier today of reconfiguring the main.py block of code Jon told me to put into a function and doing so by just calling data_manager.asset_data and data_manager.global_data, which is what you mean by "same code being used in another file", right? Would def clean up my main.py. That'll prob be my next update, and then perhaps your app idea will be a bigger update I do in the future
 
I know you mentioned the "potential" for scaling to like 100+ coins/stocks tracked in the future and this func would help with that a lot so tyty